Updating helm docs with additionalVault and ACLs refactor functionality. (#12669)
hashicorp/consul: website/content/docs/k8s/helm.mdx

        and check the name of `metadata.name`.

      - `consulClientRole` ((#v-global-secretsbackend-vault-consulclientrole)) (`string: ""`) - The Vault role for the Consul client.
        The role must be connected to the Consul client's service account.
        The role must also have a policy with read capabilities for the gossip encryption
        key defined by `global.gossipEncryption.secretName`.
        To discover the service account name of the Consul client, run
            ```shell-session
            $ helm template --show-only templates/client-serviceaccount.yaml <release-name> hashicorp/consul
            ```
        and check the name of `metadata.name`.

      - `consulSnapshotAgentRole` ((#v-global-secretsbackend-vault-consulsnapshotagentrole)) (`string: ""`) - <EnterpriseAlert inline /> The Vault role for the Consul client snapshot agent.
        The role must be connected to the Consul client snapshot agent's service account.
        The role must also have a policy with read capabilities for the snapshot agent config

>         defined by `client.snapshotAgent.configSecret.secretName`.

Reviewer: ```suggestion
        defined by the `client.snapshotAgent.configSecret.secretName` value.
```

        To discover the service account name of the Consul client, run
            ```shell-session
            $ helm template --show-only templates/client-snapshot-agent-serviceaccount.yaml --set client.snapshotAgent.enabled=true <release-name> hashicorp/consul
            ```
        and check the name of `metadata.name`.

      - `manageSystemACLsRole` ((#v-global-secretsbackend-vault-managesystemaclsrole)) (`string: ""`) - A Vault role for the Consul `server-acl-init` job, which manages setting ACLs so that clients and components can obtain ACL tokens.
        The role must be connected to the `server-acl-init` job's service account.
        The role must also have a policy with read and write capabilities for the bootstrap, replication or partition tokens
        To discover the service account name of the `server-acl-init` job, run
            ```shell-session
            $ helm template --show-only templates/server-acl-init-serviceaccount.yaml \
                  --set global.acls.manageSystemACLs=true <release-name> hashicorp/consul
            ```
        and check the name of `metadata.name`.

>       - `adminPartitionsRole` ((#v-global-secretsbackend-vault-adminpartitionsrole)) (`string: ""`) - <EnterpriseAlert inline /> A Vault role to allow Kubernetes job that creates a Consul partition for this Helm chart (`partition-init`)
>         to read Vault secret for the partition ACL token.

Reviewer: ```suggestion
      - `adminPartitionsRole` ((#v-global-secretsbackend-vault-adminpartitionsrole)) (`string: ""`) - <EnterpriseAlert inline /> A Vault role to allow a Kubernetes job create a Consul partition for this Helm chart deployment (`partition-init`). This role allows Consul to read Vault secret for the respective partition ACL token.
```
Is this correct? I'm still a bit lost here 😅

        - gossip encryption key defined by `global.gossipEncryption.secretName`.
        To discover the service account name of the Consul client, run
            ```shell-session

>             $ helm template --show-only templates/client-serviceaccount.yaml <release-name> hashicorp/consul

Reviewer: This is not rendering correctly. You might have to play with the indentation.

![image](https://user-images.githubusercontent.com/29551334/161106033-178b6adc-f6ee-4d2d-9b90-0ad286b73c12.png)

Author: How are you able to preview this?  I would love to also be able to do this myself.

Reviewer: I added a comment in the PR that explains this. But yeah, it's still breaking 😢

Reviewer 2: Maybe it needs a newline before and after? or maybe it's wrong indentation?

Author: @karl-cardenas-coding can you elaborate exactly what should happen here in terms of the visual we should see? When I look at this [preview](https://consul-d50kyx45z-hashicorp.vercel.app/docs/k8s/helm), all code blocks match the indentation of the lines above them and this one seems to as well.  Am I trying to widen this code block?

Author: oh sorry.  I thought the indentation was wrong.  I see it now.  :face-palm.
the markdown is showing up in the rendered code block.

            ```
        and check the name of `metadata.name`.

      - `consulSnapshotAgentRole` ((#v-global-secretsbackend-vault-consulsnapshotagentrole)) (`string: ""`) - <EnterpriseAlert inline /> The Vault role for the Consul client snapshot agent.

>         The role must be connected to the Consul client snapshot agent's service account and

Reviewer: We should break this up into two sentences.  Is there a way we could simplify the second sentence? The second sentence confused me.

            ```
        and check the name of `metadata.name`.

      - `manageSystemACLsRole` ((#v-global-secretsbackend-vault-managesystemaclsrole)) (`string: ""`) - A Vault role to allow Kubernetes job that manages ACLs for this Helm chart (`server-acl-init`)
        to read and update Vault secrets for the Consul's bootstrap, replication or partition tokens.
        This role must be bound the `server-acl-init`'s service account.
        To discover the service account name of the `server-acl-init` job, run
            ```shell-session
            $ helm template --show-only templates/server-acl-init-serviceaccount.yaml \
                  --set global.acls.manageSystemACLs=true <release-name> hashicorp/consul
            ```
        and check the name of `metadata.name`.

      - `adminPartitionsRole` ((#v-global-secretsbackend-vault-adminpartitionsrole)) (`string: ""`) - <EnterpriseAlert inline /> A Vault role to allow Kubernetes job that creates a Consul partition for this Helm chart (`partition-init`)
        to read Vault secret for the partition ACL token.
        This role must be bound the `partition-init`'s service account.

>         To discover the service account name of the `partition-init` job, run with Helm values for the client cluster:

Reviewer: Let's revisit this paragraph. I'm not sure I understand what this role is doing.  It seems like it's used for a Kubernetes job that creates the admin partition.  But then the next sentence confuses me "to read Vault secret for the partition ACL token." Is this part of the first sentence or a new sentence?

      and create ACL tokens and policies.
      This value is ignored if `bootstrapToken` is also set.

      - `secretName` ((#v-global-acls-replicationtoken-secretname)) (`string: null`) - The name of the Kubernetes or Vault secret that holds the replication token.

      - `secretKey` ((#v-global-acls-replicationtoken-secretkey)) (`string: null`) - The key within the Kubernetes or Vault secret that holds the replication token.

    - `partitionToken` ((#v-global-acls-partitiontoken)) - partitionToken references a Vault secret containing the ACL token to be used in non-default partitions.

>       This value should only be provided in the default partition and only when setting
>       `global.secretsBackend.vault.enabled` to true.

Reviewer: ```suggestion
      This value should only be provided in the default partition and only when setting
      the `global.secretsBackend.vault.enabled` value to true.
```

      - `secretKey` ((#v-global-acls-replicationtoken-secretkey)) (`string: null`) - The key of the Kubernetes or Vault secret.

    - `partitionToken` ((#v-global-acls-partitiontoken)) - partitionToken references a Vault secret containing the ACL token to be used in non-default partitions.
      This value should only be provided in the default partition and only when setting
      `global.secretsBackend.vault.enabled` to true.
      We will use the value of the secret stored in Vault to create an ACL token in Consul with the value of the
      secret as the secretID for the token.

>       In non-default, partitions set this secret as the `bootstrapToken`.

Reviewer: "We" is only used where we are making recommendations. Otherwise we use "you" when referring to the user. Is this something that the user will do or that Consul will do on behalf of the user? If the latter, consider using "Consul will ..."

>       - `secretName` ((#v-global-acls-partitiontoken-secretname)) (`string: null`) - The name of the path of the secret in Vault.

Reviewer: ```suggestion
      - `secretName` ((#v-global-acls-partitiontoken-secretname)) (`string: null`) - The path name of the secret in Vault.
```

    - `replicas` ((#v-client-snapshotagent-replicas)) (`integer: 2`) - The number of snapshot agents to run.

    - `configSecret` ((#v-client-snapshotagent-configsecret)) - A Kubernetes or Vault secret that should be manually created to contain the entire
      config to be used on the snapshot agent.
      This is the preferred method of configuration since there are usually storage
      credentials present. Please see Snapshot agent config (https://consul.io/commands/snapshot/agent#config-file-options)
      for details.

>       - `secretName` ((#v-client-snapshotagent-configsecret-secretname)) (`string: null`) - secretName is the name of the Kubernetes secret or Vault secret path that holds the snapshot agentconfig.

Reviewer: ```suggestion
      - `secretName` ((#v-client-snapshotagent-configsecret-secretname)) (`string: null`) -  The name of the Kubernetes secret or Vault secret path that holds the snapshot agent config.
```
None of the other definitions start with the parameter name.

>       - `secretKey` ((#v-client-snapshotagent-configsecret-secretkey)) (`string: null`) - secretKey is the key within the Kubernetes secret or Vault secret key that holds the snapshot agentconfig.

Reviewer: ```suggestion
      - `secretKey` ((#v-client-snapshotagent-configsecret-secretkey)) (`string: null`) - The key within the Kubernetes secret or Vault secret key that holds the snapshot agent config.
```

          "annotation-key": "annotation-value"
        ```

>   - `resources` ((#v-apigateway-resources)) (`map`) - Resource settings for api gateway pods.

Reviewer: ```suggestion
  - `resources` ((#v-apigateway-resources)) (`map`) - The resource settings for api gateway pods.
```

>   - `initCopyConsulContainer` ((#v-apigateway-initcopyconsulcontainer)) (`map`) - Resource settings for the `copy-consul-bin` init container.

Reviewer: ```suggestion
  - `initCopyConsulContainer` ((#v-apigateway-initcopyconsulcontainer)) (`map`) - The resource settings for the `copy-consul-bin` init container.
```
